ValveKeyValue/ValveKeyValue/StringTable.cs

```diff
@@ -0,0 +1,58 @@
+﻿namespace ValveKeyValue
+{
+    public sealed class StringTable
+    {
+        public StringTable(Memory<string> values)
+        {
+            this.lookup = values;
+        }
+
+        readonly Memory<string> lookup;
+        Dictionary<string, int> reverse;
+
+        public string this[int index]
+        {
+            get
+            {
+                if (index < 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(index), "Index must be non-negative.");
+                }
+
+                if (index >= lookup.Length)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(index), index, "Index must be less than the number of strings in the table.");
+                }
+
+                return lookup.Span[index];
+            }
+        }
+
+        public int IndexOf(string value)
+        {
+            if (reverse is null)
+            {
+                throw new InvalidOperationException("String table has not been prepared for serialization.");
+            }
+
+            return reverse[value];
+        }
+
+        public void PrepareForSerialization()
+        {
+            if (reverse is not null)
+            {
+                return;
```

Comment:
Is this gonna cause issues if you serialize, change the kv and try to serialize again?

Actually not seeing a way for stringtable to be populated by the seializer.

Reply:
oh that's an interesting idea, should we dynamically populate a string table?

Reply:
If we want to support serialization, we kind of have to.

Reply:
We also need a public accessor to the list if that happens, because you need to store it somehow